source/slang/slang-end-to-end-request.cpp

```diff
@@ -1576,8 +1576,17 @@ SlangResult EndToEndCompileRequest::compile()
     auto reflectionPath = getOptionSet().getStringOption(CompilerOptionName::EmitReflectionJSON);
     if (reflectionPath.getLength() != 0)
     {
+        auto reflection = this->getReflection();
+        if (!reflection)
+        {
+            getSink()->diagnose(
+                SourceLoc(),
+                Diagnostics::cannotEmitReflectionWithoutTarget,
+                "Cannot emit reflection JSON: no compilation target available");
```

Comment:
Please delete the string argument here, since it isn't actually being used by the diagnostic emit logic.